tests/sbiutils_test.py

```diff
@@ -406,30 +407,79 @@ def test_kde(bandwidth, transform, sample_weights):
 
 
 @pytest.mark.parametrize(
-    "z_x", [True, False, None, "none", "independent", "structured"]
+    "z_x",
+    [True, False, None, "none", "independent", "structured", "logit", "invalid_value"],
 )
 @pytest.mark.parametrize(
-    "z_theta", [True, False, None, "none", "independent", "structured"]
+    "z_theta",
+    [True, False, None, "none", "independent", "structured", "logit", "invalid_value"],
+)
+def test_z_score_parser(z_x, z_theta):
+    if z_theta == "invalid_value":
+        with pytest.raises(ValueError, match="Invalid z-scoring option."):
+            z_score_parser(z_theta)
+
+    if z_x == "invalid_value":
+        with pytest.raises(ValueError, match="Invalid z-scoring option."):
+            z_score_parser(z_x)
+
+    if z_x != "invalid_value" and z_theta != "invalid_value":
+        result_x = z_score_parser(z_x)
+        result_theta = z_score_parser(z_theta)
+
+        assert result_x is not None, f"z_score_parser({z_x}) returned None"
+        assert result_theta is not None, f"z_score_parser({z_theta}) returned None"
+
+
+@pytest.mark.parametrize(
+    "z_x", [True, False, None, "none", "independent", "structured", "logit"]
+)
+@pytest.mark.parametrize(
+    "z_theta", [True, False, None, "none", "independent", "structured", "logit"]
 )
 @pytest.mark.parametrize("builder", [likelihood_nn, posterior_nn, classifier_nn])
 def test_z_scoring_structured(z_x, z_theta, builder):
     """
     Test that z-scoring string args don't break API.
     """
     # Generate some signals for test.
-    t = torch.arange(0, 1, 0.001)
+    t = torch.arange(0, 1, 0.1)
     x_sin = torch.sin(t * 2 * torch.pi * 5)
     t_batch = torch.stack([(x_sin * (i + 1)) + (i * 2) for i in range(10)])
 
+    num_dim = t_batch.shape[1]
+    x_dist = BoxUniform(
+        low=-2 * torch.ones(num_dim), high=2 * torch.ones(num_dim)
+    )  # Normal(0, 1) if (z_x == "logit" or z_theta == "logit") else None
+
     # API tests
+    # TODO: Test breaks at "mnle"
```

Comment:
`mnle` has discrete `x`. But the parameters should still be continuous, and you don't change the condition standardizing net as far as I can tell? Let me know if you need help debugging

Reply:
Potentially since there is a separate `build` function for `build_mnle` compared to the other flows (it is also in a different file which may have cuased you to miss it